gluten-core/src/main/scala/org/apache/gluten/expression/ExpressionMappings.scala

```diff
@@ -338,8 +338,11 @@ object ExpressionMappings {
 
   def expressionsMap: Map[Class[_], String] = {
     val blacklist = GlutenConfig.getConf.expressionBlacklist
-    val supportedExprs = defaultExpressionsMap ++
-      expressionExtensionTransformer.extensionExpressionsMapping
+
+    var supportedExprs = defaultExpressionsMap
+    if (expressionExtensionTransformer != null) {
```

Comment:
How about giving `expressionExtensionTransformer` a initial value `DefaultExpressionExtensionTransformer` ? it seems quite risk to check if not null at everywhere..

Reply:
@ulysses-you It's strange that the `ExpressionUtil.extendedExpressionTransformer` method is returning null. I'm not sure why this is happening. It should be return the DefaultExpressionExtensionTransformer. Do you have any insights?

Reply:
you mean `ExpressionUtil.extendedExpressionTransformer` returns null but not `DefaultExpressionExtensionTransformer()` when `extendedExpressionTransformer` is empty ?

Reply:
@zzcclp Yes.